src/integrationTest/cloudformation/intellisense.test.ts

```diff
@@ -0,0 +1,116 @@
+        dir++
+
+        // Stub showSaveDialog so we can create a starter cloudformation template
+        sinon.stub(vscode.window, 'showSaveDialog').returns(Promise.resolve(testFileUri))
+    })
+
+    afterEach(async function () {
+        await fs.remove(testDir)
+        sinon.restore()
+    })
+
+    it('autocompletes a cloudformation file', async function () {
+        await vscode.commands.executeCommand('aws.cloudFormation.newTemplate')
+        await waitUntil(
+            () => {
+                return Promise.resolve(globals.schemaService.isMapped(testFileUri))
+            },
+            { timeout: MAPPING_SCHEMA_TIMEOUT, interval: MAPPING_SCHEMA_INTERVAL, truthy: true }
+        )
+
+        const completions = await getCompletionItems(testFileUri, new vscode.Position(4, 0))
+        if (!completions) {
+            assert.fail('Expected completion items in cloudformation file')
+        }
+        const filterTags = completions.items.filter(completion => !completion.label.startsWith('!'))
+        assert.notStrictEqual(filterTags.length, 0, 'Expected more than 0 completion items')
+    })
+
+    it('hovers a cloudformation file', async function () {
+        await vscode.commands.executeCommand('aws.cloudFormation.newTemplate')
+        await waitUntil(
+            () => {
+                return Promise.resolve(globals.schemaService.isMapped(testFileUri))
+            },
+            { timeout: MAPPING_SCHEMA_TIMEOUT, interval: MAPPING_SCHEMA_INTERVAL, truthy: true }
+        )
+
+        const hovers = await getHoverItems(testFileUri, new vscode.Position(0, 0))
+        if (!hovers) {
+            assert.fail('Expected hover items in cloudformation file')
+        }
+        assert.notStrictEqual(hovers.length, 0, 'Expected more than 0 hover items')
+    })
+
+    it('autocompletes a sam file', async function () {
+        await vscode.commands.executeCommand('aws.sam.newTemplate')
+        await waitUntil(
+            () => {
+                return Promise.resolve(globals.schemaService.isMapped(testFileUri))
+            },
+            { timeout: MAPPING_SCHEMA_TIMEOUT, interval: MAPPING_SCHEMA_INTERVAL, truthy: true }
+        )
+
+        const completions = await getCompletionItems(testFileUri, new vscode.Position(4, 0))
+        if (!completions) {
+            assert.fail('Expected completion items in sam file')
+        }
+        const filterTags = completions.items.filter(completion => !completion.label.startsWith('!'))
+        assert.notStrictEqual(filterTags.length, 0, 'Expected more than 0 completion items')
+    })
+
+    it('hovers a sam file', async function () {
+        await vscode.commands.executeCommand('aws.sam.newTemplate')
+        await waitUntil(
+            () => {
+                return Promise.resolve(globals.schemaService.isMapped(testFileUri))
+            },
+            { timeout: MAPPING_SCHEMA_TIMEOUT, interval: MAPPING_SCHEMA_INTERVAL, truthy: true }
```

Comment:
Unless `isMapped` is an expensive operation, the default of 500ms is fine. So we don't need this MAPPING_SCHEMA_INTERVAL global. Avoiding unnecessary symbols avoids questions for future developers about the intentions of the codebase.
```suggestion
            { timeout: MAPPING_SCHEMA_TIMEOUT }
```